freud/cluster.pyx

```diff
@@ -206,45 +207,122 @@ cdef class ClusterProperties(_Compute):
                 :class:`freud.locality.NeighborQuery.from_system`.
             cluster_idx ((:math:`N_{points}`,) :class:`np.ndarray`):
                 Cluster indexes for each point.
+            masses (:math:`\left(N, 3\right)` :class:`numpy.ndarray`):
+                Masses corresponding to each point, defaulting to 1 if not
+                provided or :code:`None` (Default value = :code:`None`).
         """
         cdef freud.locality.NeighborQuery nq = \
             freud.locality.NeighborQuery.from_system(system)
         cluster_idx = freud.util._convert_array(
             cluster_idx, shape=(nq.points.shape[0], ), dtype=np.uint32)
         cdef const unsigned int[::1] l_cluster_idx = cluster_idx
-        self.thisptr.compute(
-            nq.get_ptr(),
-            <unsigned int*> &l_cluster_idx[0])
+
+        cdef float* l_masses_ptr = NULL
+        cdef float[::1] l_masses
+        if masses is not None:
+            l_masses = freud.util._convert_array(masses, shape=(len(masses), ))
+            l_masses_ptr = &l_masses[0]
+
+        self.thisptr.compute(nq.get_ptr(),
+                             <unsigned int*> &l_cluster_idx[0],
+                             l_masses_ptr)
         return self
 
     @_Compute._computed_property
     def centers(self):
-        """(:math:`N_{clusters}`, 3) :class:`numpy.ndarray`: The centers of
-        mass of the clusters."""
+        r"""(:math:`N_{clusters}`, 3) :class:`numpy.ndarray`: The geometric centers
+        of the clusters, independent of mass and defined as
+
+        .. math::
+
+            \mathbf{C}_m = \frac{1}{N} \sum_{i=0}^{N} \mathbf{r_i}
```

Comment:
What does the `m` subscript refer to here?

Reply:
Oops, typo. It should be `g` for geometric, as opposed to mass, `m`.